homeassistant/components/amcrest/config_flow.py

```diff
@@ -0,0 +1,233 @@
+"""Config flow to configure Amcrest devices."""
+
+import logging
+
+from amcrest import AmcrestError, LoginError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_AUTHENTICATION,
+    CONF_BINARY_SENSORS,
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SCAN_INTERVAL,
+    CONF_SENSORS,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+
+from .amcrest_checker import AmcrestChecker
+from .const import (
+    AUTHENTICATION_LIST,
+    CONF_BINARY_SENSOR_AUDIO_DETECTED,
+    CONF_BINARY_SENSOR_AUDIO_DETECTED_POLLED,
+    CONF_BINARY_SENSOR_MOTION_DETECTED,
+    CONF_BINARY_SENSOR_MOTION_DETECTED_POLLED,
+    CONF_BINARY_SENSOR_ONLINE,
+    CONF_CONTROL_LIGHT,
+    CONF_EVENTS,
+    CONF_FFMPEG_ARGUMENTS,
+    CONF_RESOLUTION,
+    CONF_SENSOR_PTZ_PRESET,
+    CONF_SENSOR_SDCARD,
+    CONF_STREAM_SOURCE,
+    DEFAULT_AUTHENTICATION,
+    DEFAULT_CONTROL_LIGHT,
+    DEFAULT_FFMPEG_ARGUMENTS,
+    DEFAULT_NAME,
+    DEFAULT_PORT,
+    DEFAULT_RESOLUTION,
+    DEFAULT_SCAN_INTERVAL,
+    DEFAULT_STREAM_SOURCE,
+    DEVICES,
+    DOMAIN,
+    RESOLUTION_LIST,
+    STREAM_SOURCE_LIST,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AmcrestFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Amcrest config flow."""
+
+    VERSION = 2
```

Comment:
You're introducing a new one, so version = 1
```suggestion
    VERSION = 1
```